snakemake/subworkflow.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+from snakemake.logging import logger
+from snakemake.exceptions import WorkflowError
+
+
+def subworkflow(
+    path,
+    basedir,
+    input,
+    output,
+    params,
```

Comment:
how are params passed to rules in a subworkflow? Are they exposed in the config dict?

Reply:
The `params` directive could generally be used to modify the sub-workflows setup (as `params. configfile` and `params.workdir` currently do).
Furthermore, `params.config` could be merged into the sub-workflows configuration, `params. restart_times` could set sub-workflow specific restart times, etc.